### PR TITLE
fix: omit proxy path prefix from socket.io base path

### PR DESCRIPTION
**Pull request by bvenreply**
#### What it does

Fixes an issue causing socket.io to use the wrong basepath when accessing the backend from a reverse proxy. #13055

#### How to test

This can be tested by running the browser example and accessing via a local nginx proxy:

- Build/run the browser example with `yarn && yarn browser build && yarn browser start` (Theia will run on port 3000)
- Run nginx using the following `nginx.conf`
  ```nginx
  http {
      server { 
          listen 3300;
  
          location ~* /proxy/(.*)? {
              proxy_set_header Host $host;
              proxy_set_header X-Real-IP $remote_addr;
              proxy_set_header X-Forwarded-For $remote_addr;
              proxy_set_header Upgrade $http_upgrade;
              proxy_set_header Connection "Upgrade";
              proxy_ssl_verify off;
  
              if ($args) {
                  proxy_pass http://127.0.0.1:3000/$1?$args;
              }
  
              proxy_pass http://127.0.0.1:3000/$1;
          }
      }
  }
  events {}
  ```
- Access theia via http://localhost:3300/proxy/

#### Follow-ups

Nothing major that i can see.

The first `path` argument [here](https://github.com/eclipse-theia/theia/commit/5107942562e2c2c9efb2afeb39f027ab8f0d27f5#diff-4bce5ab557626597cf9f000f9841f01f8ebf30c1bec36544fc4074cfe922711dR122) is now redundant; probably not really worth worrying about considering the severity of the issue.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by msujew — APPROVED** [review]
Thanks, looks good to me. Seems like my recent changes in https://github.com/eclipse-theia/theia/pull/12618 accidentally broke this.

**Comment by msujew** [comment]
Before we can merge, you'll need to sign the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php) though. Please do that with the same email that was used to create the commit.

**Comment by bvenreply** [comment]
> Before we can merge, you'll need to sign the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php) though. Please do that with the same email that was used to create the commit.

Done @msujew

**Comment by msujew** [comment]
@bvenreply Seems like the ECA validation still doesn't go through. Have you signed the ECA with `benia@protonmail.com`? That is the email associated with the commit.

**Comment by bvenreply** [comment]
@msujew I have:

image

Is the revalidation usually instant?

**Comment by msujew** [comment]
Not instant, but it's weird it's taking that long. We'll probably see tomorrow ¯\_(ツ)_/¯

**Comment by msujew** [comment]
I force pushed on your branch to fix the ECA check. For some reason it didn't work without a new commit. I'll merge this once the CI passes.

**Pull request merged**
